docs/examples_grating_coupler.rst

```diff
@@ -0,0 +1,365 @@
+    if etch_frac != 1:
+        stack.append(
+            optplan.GdsMaterialStackLayer(
+                foreground=optplan.Material(index=optplan.ComplexNumber(real=2.0,imag=0.0))
+                background=optplan.Material(mat_name="Air"),
+                gds_layer=[LAYER_SILICON_NONETCHED, 0],
+                extents=[0, wg_thickness * (1 - etch_frac)],
+            ))
+    stack.append(
+        optplan.GdsMaterialStackLayer(
+            foreground=optplan.Material(index=optplan.ComplexNumber(real=2.0,imag=0.0))
+            background=optplan.Material(mat_name="Air"),
+            gds_layer=[LAYER_SILICON_ETCHED, 0],
+            extents=[wg_thickness * (1 - etch_frac), wg_thickness],
+        ))
+
+In addition, we change the background material to be :code:`"Air"` as our grating is air-cladded.
+
+.. code-block:: python
+
+    mat_stack = optplan.GdsMaterialStack(
+        # Any region of the simulation that is not specified is filled with
+        # oxide.
+        background=optplan.Material(mat_name="Air"),
+        stack=stack,
+    )
+
+
+Grating parameters
+******************
+
+We set the partial etch depth earlier, but to re-iterate, we can adjust this value in the :code:`run_opt` function in the arguments to the :code:`create_sim_space` call:
+
+.. code-block:: python
+
+    sim_space = create_sim_space(
+            "sim_fg.gds",
+            "sim_bg.gds",
+            grating_len=grating_len,
+            box_thickness=3000,
+            wg_thickness=wg_thickness,
+            etch_frac=0.6,
+            wg_width=wg_width)
+    
+We see reference to grating_len here, and accordingly this variable can be adjusted as well. This is set at the bottom of the example file in the :code:`__main__` function call:
+
+.. code-block:: python
+
+    if __name__ == "__main__":
+        import argparse
+    
+        parser = argparse.ArgumentParser()
+        parser.add_argument(
+            "action",
+            choices=("run", "view", "view_quick", "resume", "gen_gds"),
+            help="Must be either \"run\" to run an optimization, \"view\" to "
+            "view the results, \"resume\" to resume an optimization, or "
+            "\"gen_gds\" to generate the grating GDS file.")
+        parser.add_argument(
+            "save_folder", help="Folder containing optimization logs.")
+    
+        grating_len = 12000
+        wg_width = 12000
+    
+
+Source parameters
+*****************
+
+Source details are defined in the function :code:`create_objective`
+
+In this function, wavelength is set by simply adjusting the :code:`wlen` variable in the :code:`create_objective` function.
+
+.. code-block:: python
+
+    wlen = 1300
+
+
+.. note:: 
+
+    Another location where wavelength is referenced is for plotting the permitvitty for visualization. If desired, adjust the wavelength argument in the :code:`create_sim_space` function at the bottom:
+    
+    .. code-block:: python
+
+        if visualize:
```

Comment:
I think it's worth having a section mentioning how to turn on the `visualize` flag just to see what the permittivity distirbution looks like after you modify the material stack.